addons/website/models/ir_module_module.py

```diff
@@ -159,7 +159,7 @@ def _update_records(self, model_name, website):
                 if not find and model_name == 'ir.attachment':
                     # In master, a unique constraint over (theme_template_id, website_id)
                     # will be introduced, thus ensuring unicity of 'find'
-                    find = rec.copy_ids.search([('key', '=', rec.key), ('website_id', '=', website.id), ("original_id", "=", False)])
+                    find = rec.copy_ids.search([('key', '=', rec.key), ('website_id', '=', website.id), ("original_id", "=", False), ("store_fname", "=", False)])
```

Comment:
Hello, sorry for the delay, I was on vacation. The explanation seems good but I am confused by some things:

- Why is it targeting 17.2? i.e. there should be a reference to a commit that changed something in 17.2 that created the issue in the commit message
- Should there be a reference to a ticket or whatever in the commit message ?
- I am not sure to understand that whole `if`. `# if module B override attachment from dependence A, we update it` what case is that ? Why is the line above that filters inside `copy_ids` o2m not enough? Maybe that actually is the root of the issue? That `rec.copy_ids.search` does not search inside the `copy_ids` records?